### DISCUSSION TITLE
Question: loading third party libraries in hooks

### DISCUSSION THREAD

**Discussion by agzam** [clj-kondo/clj-kondo#1449] [Q&A]
This may sound like wishing to capture a shooting star, but I'm curious if this is even remotely possible.

I was playing with custom hooks and `:analyze-call` and I thought:

Is there any practical possibility for loading namespaces from libs used in the project? Given that `.clj-kondo/config.edn` is within a project?

I don't have a specific problem in mind, just curious of the limitations, that are (no doubt) present.

**Comment by borkdude** [comment]
No, this isn't possible. The options in hooks are restricted on purpose for both performance and safety. Questions can be posted to the Discussions part of this repo.

**Reply by borkdude** [reply]
@agzam If you're looking for another option similar to `:analyze-call`, try `:macroexpand` which is documented in the same documentation page.
